src/YoutubeAtom.tsx

```diff
@@ -300,21 +346,27 @@ export const YoutubeAtom = ({
                 />
             )}
 
-            {(overrideImage || posterImage) && (
+            {showOverlay && (
                 <div
                     daya-cy="youtube-overlay"
+                    data-testid="youtube-overlay"
                     onClick={() => {
                         setHasUserLaunchedPlay(true);
-                        player.current && player.current.playVideo();
+                        iframeSrc &&
+                            player.current &&
+                            player.current.playVideo();
                     }}
                     onKeyDown={(e) => {
                         const spaceKey = 32;
                         const enterKey = 13;
                         if (e.keyCode === spaceKey || e.keyCode === enterKey) {
```

Comment:
Outside the scope of this PR, but [`e.keyCode` is deprecated](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code)

Reply:
Yeah, my IDE is saying the same thing. I'll have a quick look at changing this, it's easier to do these things while you're in the code

Reply:
```suggestion
                        if (e.code === "Space" || e.code === "Enter") {
```